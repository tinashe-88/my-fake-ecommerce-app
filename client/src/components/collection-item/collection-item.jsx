import React from 'react'
import { connect } from 'react-redux'

import {addItem} from  '../../redux/cart/cart.actions'

import {
  CollectionItemContainer,
  AddButtonContainer,
  ImageContainer,
  CollectionFooterContainer,
  NameContainer,
  PriceContainer
} from './collection-item.styles'

export const CollectionItem = ({ item, addItem }) => {
  const { name, price, imageUrl } = item

  return (
    <CollectionItemContainer>
      <ImageContainer imageUrl={imageUrl}/>
      <CollectionFooterContainer>
        <NameContainer>{name}</NameContainer>
        <PriceContainer>{price}</PriceContainer>
      </CollectionFooterContainer>
      <AddButtonContainer onClick={() => addItem(item)} inverted>
        Add To Cart
      </AddButtonContainer>
    </CollectionItemContainer>
  )
}


const mapDispatchToProps = dispatch => ({
  addItem: item => dispatch(addItem(item))
})

export default connect(null, mapDispatchToProps)(CollectionItem)