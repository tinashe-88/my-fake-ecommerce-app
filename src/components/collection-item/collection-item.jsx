import React from 'react'

import {
  CollectionItemContainer,
  AddButton,
  BackgroundImage,
  CollectionFooterContainer,
  NameContainer,
  PriceContainer
} from './collection-item.styles'

const CollectionItem = ({ id, name, price, imageUrl }) => (
  <CollectionItemContainer>
    <BackgroundImage
      style={{
        backgroundImage: `url(${imageUrl})`
      }}
    />
    <CollectionFooterContainer>
      <NameContainer>{name}</NameContainer>
      <PriceContainer>{price}</PriceContainer>
    </CollectionFooterContainer>
  </CollectionItemContainer>
)

export default CollectionItem