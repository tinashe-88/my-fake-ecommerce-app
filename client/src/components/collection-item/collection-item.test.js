import React from 'react'
import { shallow } from 'enzyme'

import { CollectionItem } from './collection-item'

describe('', () => {
  let wrapper
  let mockAddItem

  const imageUrl = 'www.imageUrl.com'
  const mockName = 'blue briefs'
  const mockPrice = 240

  beforeEach(() => {
    mockAddItem = jest.fn()

    const mockProps = {
      item: {
        imageUrl: imageUrl,
        name: mockName,
        price: mockPrice
      },
      addItem: mockAddItem
    }

    wrapper = shallow(<CollectionItem {...mockProps}/>)
  })

  it('Should render CollectionItem component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should render imageUrl as a prop in ImageContainer', () => {
    expect(wrapper.find('ImageContainer').prop('imageUrl')).toBe(imageUrl)
  })

  it('Should call addItem when AddButtonContainer is clicked', () => {
    wrapper.find('AddButtonContainer').simulate('click')
    expect(mockAddItem).toHaveBeenCalled()
  })

  it('Should render name prop in NameContainer', () => {
    expect(wrapper.find('NameContainer').text()).toBe(mockName)
  })

  it('Should render price prop in PriceContainer', () => {
    const price = parseInt(wrapper.find('PriceContainer').text())
    expect(price).toBe(mockPrice)
  })
})