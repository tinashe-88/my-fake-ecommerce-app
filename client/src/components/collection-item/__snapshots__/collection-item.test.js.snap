// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render CollectionItem component 1`] = `
<styled.div>
  <ImageContainer
    imageUrl="www.imageUrl.com"
  />
  <styled.div>
    <NameContainer>
      blue briefs
    </NameContainer>
    <PriceContainer>
      240
    </PriceContainer>
  </styled.div>
  <AddButtonContainer
    inverted={true}
    onClick={[Function]}
  >
    Add To Cart
  </AddButtonContainer>
</styled.div>
`;
